Library/Sender.ts

```diff
@@ -97,6 +100,24 @@ class Sender {
             }
         };
 
+        let authHandler = this._getAuthorizationHandler ? this._getAuthorizationHandler() : null;
+        if (authHandler) {
+            try {
+                // Add bearer token
+                await authHandler.addAuthorizationHeader(options);
+            }
+            catch (authError) {
```

Comment:
I believe we want to do different things (retry, drop, etc.) based off the type of exception that is thrown when getting the token. [Python](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/README.md#error-handling) throws these types of exceptions (I believe Java is similar). For `CredentialUnavailableError ` we probably do not want to retry. For `ClientAuthenticationError` we should investigate which cases make sense to retry.

Reply:
If we do not retry telemetry will be lost on the spot in that case, even if customer configuration is wrong we should try to store the telemetry in disk, what about AAD outages, maybe CredentialUnavailableError is triggered in that case?